service/client/client.js

```diff
@@ -717,6 +717,14 @@ export default class Client {
             data = await response.json();
         } else {
             data = await response.text();
+            if (isResponseInvalid(data)) {
```

Comment:
Because the only thing that is actually validating that is a JSON is the `Content-Type` so in case the content type is not set to json we wan't to validate it